src/bindings.cpp

```diff
@@ -193,6 +194,65 @@ extern "C"
         return (PlayerErrors)noError;
     }
 
+    /// Change the playback device.
+    ///
+    /// [deviceID] the device ID. -1 for default OS output device.
+    FFI_PLUGIN_EXPORT enum PlayerErrors changeDevice(int deviceID)
+    {
+        if (player.get() == nullptr)
+            return backendNotInited;
+
+        return player.get()->changeDevice(deviceID);
+    }
+
+    /// List playback devices.
+    FFI_PLUGIN_EXPORT void listPlaybackDevices(
+        char **devicesName,
+        int **deviceId,
+        int **isDefault,
+        int *n_devices)
+    {
+        std::vector<PlaybackDevice> d = player.get()->listPlaybackDevices();
+
+        int numDevices = 0;
+        for (int i = 0; i < (int)d.size(); i++)
+        {
+            bool hasSpecialChar = false;
+            /// check if the device name has some strange chars (happens on Linux)
+            for (int n = 0; n < 5; n++)
+            {
+                if (d[i].name[n] < 0x20)
+                    hasSpecialChar = true;
+            }
+            if (strlen(d[i].name) <= 5 || hasSpecialChar)
+                continue;
```

Comment:
Please add a comment that explains why we're skipping this device based on this condition.

Reply:
A few lines above I wrote a comment and added few lines now. It happened that some results had the name composed of non-text ASCII characters. Characters with values ​​<0x20 (blank space). For some time now it no longer happens on my Linux, it was probably some issue in the OS, but I preferred to leave this check anyway.

Reply:
I'm pretty sure those devices are junk data from `/dev/snd` I got a while back. I don't know how miniaudio gets those devices but in my opinion, they are not valid.

Added some words in the comment.